From c64899ab6c2986b90fe6ee050dab552fe3bd2386 Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Thu, 21 Aug 2014 13:06:17 -0400
Subject: [ethernet] qlcnic: Fix usage of use_msi and use_msi_x module parameters

Message-id: <1408626422-3941-94-git-send-email-cdupuis@redhat.com>
Patchwork-id: 88608
O-Subject: [PATCH RHEL 7.1 093/138] qlcnic: Fix usage of use_msi and use_msi_x module parameters
Bugzilla: 1089352
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Shahed Shaikh <shahed.shaikh@qlogic.com>

Bug 1089352 (https://bugzilla.redhat.com/show_bug.cgi?id=1089352)

net commit id b7520d2b59b09eb284e5fc9080d13145f0a8d9fd

Once interrupts are enabled, instead of using module parameters,
use flags (QLCNIC_MSI_ENABLED and QLCNIC_MSIX_ENABLED) set by driver
to check interrupt mode.

Signed-off-by: Shahed Shaikh <shahed.shaikh@qlogic.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
index d2e9ea8..c884fc0 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
@@ -3869,7 +3869,7 @@ int qlcnic_validate_rings(struct qlcnic_adapter *adapter, __u32 ring_cnt,
   strcpy(buf, "Tx");
  }
 
- if (!qlcnic_use_msi_x && !qlcnic_use_msi) {
+ if (!QLCNIC_IS_MSI_FAMILY(adapter)) {
   netdev_err(netdev, "No RSS/TSS support in INT-x mode\n");
   return -EINVAL;
  }
-- 
1.7.1